ccm-connector/src/main/java/com/sequenceiq/cloudbreak/ccmimpl/ccmv2/CcmV2ManagementClient.java

```diff
@@ -81,6 +82,22 @@ public UnregisterAgentResponse deregisterInvertingProxyAgent(String requestId, S
                 });
     }
 
+    public List<InvertingProxyAgent> listInvertingProxyAgents(String requestId, String accountId, Optional<String> environmentCrnOpt) {
+        return getRetryTemplate().execute(
+                retryContext -> {
+                    LOGGER.debug("Listing Agents for account '{}' and environmentCrn '{}'", accountId, environmentCrnOpt);
+                    List<InvertingProxyAgent> agentList = grpcCcmV2Client.listAgents(requestId, ThreadBasedUserCrnProvider.getUserCrn(),
+                            accountId, environmentCrnOpt);
+                    LOGGER.debug("Listed Agent for account '{}' and environmentCrn '{}' list count: '{}'", accountId, environmentCrnOpt, agentList.size());
+                    return agentList;
+                },
+                retryExhausted -> {
+                    LOGGER.error("Error Listing Agents for account '{}' and environmentCrn '{}', retryCount '{}'",
+                            accountId, environmentCrnOpt, retryExhausted.getRetryCount());
+                    throw new CcmV2Exception("Error Listing CCM Agents", retryExhausted.getLastThrowable());
```

Comment:
do we sending this back to the customers?

Reply:
it is displayed on freeipa stack status

Reply:
now the message is changed in an upper level